multi-arch-builders/builder-splunk.bu

```diff
@@ -0,0 +1,123 @@
+# This butane config will do the following:
+#
+# - create a splunk user
+# - build and start a splunk forwarder container to send journald logs to splunk
+#
+# See the rhcos multiarch builder splunk documentation in the CoreOS Team Operations GitLab repo.
+# The following env variables need to be set before creating the ignition config:
+#
+# - ${DEPLOYMENT_CLIENT}
+# - ${SPLUNK_HOSTNAME}
+# - ${ITPAAS_SPLUNK_REPO}
+# - ${SPLUNK_SIDECAR_REPO}
+#
+variant: fcos
+version: 1.4.0
+passwd:
+  users:
+    - name: splunk
+      groups:
+        - systemd-journal
+storage:
+  directories:
+    - path: /home/splunk/.config
+      user:
+        name: splunk
+      group:
+        name: splunk
+    - path: /home/splunk/.config/systemd
+      user:
+        name: splunk
+      group:
+        name: splunk
+    - path: /home/splunk/.config/systemd/user
+      user:
+        name: splunk
+      group:
+        name: splunk
+    - path: /home/splunk/.config/systemd/user/default.target.wants
+      user:
+        name: splunk
+      group:
+        name: splunk
+    - path: /home/splunk/.config/containers
+      user:
+        name: splunk
+      group:
+        name: splunk
+    - path: /home/splunk/.config/containers/systemd
+      user:
+        name: splunk
+      group:
+        name: splunk
+  files:
+    - path: /var/lib/systemd/linger/splunk
+      mode: 0644
+    - path: /home/splunk/.config/containers/systemd/splunk.container
+      mode: 0644
+      user:
+        name: splunk
+      group:
+        name: splunk
+      contents:
+        inline: |
+          [Unit]
+          Description=Start the splunk container to send journald logs to splunk
+          After=build-splunk.service
+          [Container]
+          ContainerName=rhcos-multiarch-splunk
+          Image=localhost/splunkforwarder-rhcos:latest
+          # set the user to root to run the splunk process
+          # as the splunk user on the host
+          User=root
+          Volume=/var/log/journal:/var/log/journal:ro
+          Volume=/run/log/journal:/run/log/journal:ro
+          Volume=/etc/machine-id:/etc/machine-id:ro
+          # use the container host's network to properly establish
+          # a connection with the splunk server
+          Network=host
```

Comment:
Even with the comment I still don't quite understand why this is needed maybe we can get toghether and I can understand more?

Reply:
Sure we can chat more about it. I got this information from the ITPAAS internal documentation and code. The splunk team also advised me to use this option. I tried to run the container without it and it wouldn't connect to to the server.

Reply:
interesting. would definitely be nice to know why - maybe some issues with NAT?